crates/celo-revm/src/contracts/core_contracts.rs

```diff
@@ -68,44 +68,73 @@ pub fn get_revert_message(output: Bytes) -> String {
     }
 }
 
+/// Call a core contract function and return the result. It is read-only, not modifying the state.
 pub fn call<DB, INSP>(
     evm: &mut CeloEvm<DB, INSP>,
     address: Address,
     calldata: Bytes,
 ) -> Result<Bytes, CoreContractError>
 where
-    DB: Database,
+    DB: Database + DatabaseCommit,
     INSP: Inspector<CeloContext<DB>>,
 {
-    // Create checkpoint to revert changes after the call
-    let checkpoint = evm.ctx().journal().checkpoint();
-    let call_result = mutable_call(evm, address, calldata);
+    // Preserve the tx set in the evm before the call to restore it afterwards
+    let prev_tx = evm.ctx().tx().clone();
+
+    let call_result = evm.transact_system_call(address, calldata);
```

Comment:
We don't commit the state changes, but I'm not sure if there are other changes to the evm object (especially the journal) after this call. Do you know?